packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -513,22 +513,13 @@ export const useGridVirtualScroller = () => {
   );
 
   const contentSize = React.useMemo(() => {
-    // In cases where the columns exceed the available width,
-    // the horizontal scrollbar should be shown even when there're no rows.
-    // Keeping 1px as minimum height ensures that the scrollbar will visible if necessary.
-    const height = Math.max(contentHeight, 1);
-
```

Comment:
I think this code was preventing [this 0-height warning](https://github.com/mui/mui-x/blob/5ef1694752a5fe50fac5a74f6835ec7fe0779c9f/packages/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts#L334-L345) from running, so fixing it broke some tests all over the place that didn't have explicit dimensions on the grid container. I'll need to debug & update those tests before merging this PR. The test runner also doesn't seem to match the warning to which test it originated from precisely, so sometimes it marks one test as failed but the warning came from another test. Lots of fun ahead.